File: src/NzbDrone.Core/Blocklisting/Manga/MangaBlocklist.cs
using System;
using System.Collections.Generic;
using NzbDrone.Core.Datastore;

namespace NzbDrone.Core.Blocklisting.Manga
{
    // Sonarr divergence: NEW manga sibling per Phase 6 D-11 + D-19 — see DIVERGENCE.md.
    // Role-match analog: src/NzbDrone.Core/Blocklisting/Blocklist.cs.
    //
    // D-11 release-identity triple: (SourceKey, ReleaseGuid, SourceTitle) — distinct from TV's
    // (Title, Indexer, Protocol, TorrentInfoHash) quintuple. Manga has no Quality, no Protocol
    // (in v1 only DownloadProtocol.Http is in play), and no TorrentInfoHash. The triple plus
    // MangaId + ChapterIds is enough to dedup blocklist rows uniquely.
    //
    // SourceKey is the per-Phase-3 indexer canonical key (e.g. "MangaDex", "comix.to" — NOT the
    // user-facing display name). ReleaseInfo.Indexer carries this value at the wire layer per
    // existing ChapterHistoryService.Handle precedent (Plan 06-03 line 141).
    //
    // Phase 8 cleanup: collapse with src/NzbDrone.Core/Blocklisting/Blocklist.cs when Tv/ deletes.
    public class MangaBlocklist : ModelBase
    {
        public int MangaId { get; set; }
        public List<int> ChapterIds { get; set; } = new List<int>();
        public string SourceTitle { get; set; }
        public string SourceKey { get; set; }
        public string ReleaseGuid { get; set; }
        public string ReleaseInfoJson { get; set; }
        public DateTime Date { get; set; }
        public string Reason { get; set; }
        public string Source { get; set; }
    }
}

File: frontend/src/typings/MangaBlocklist.ts
// Sonarr divergence: NEW manga sibling per Phase 7 D-07 — see DIVERGENCE.md.
// Role-match analog: frontend/src/typings/Blocklist.ts.
//
// Manga sibling preserves: ModelBase + URL-shaped React Query key
// (`['/manga/blocklist']` per Plan 07-02 contract).
//
// Manga sibling diverges from Blocklist:
//   * No quality / languages / protocol / torrentInfoHash (manga has no quality
//     model per Phase 5 D-04; v1 only DownloadProtocol.Http per Phase 4 D-10;
//     D-11 release identity = (sourceKey, releaseGuid, sourceTitle) triple).
//   * Adds chapterIds array (the set of chapters this blocklist row covers).
//   * Field name is `reason` (NOT `message` like Blocklist) — mirrors backend
//     `MangaBlocklistResource.Reason`.
//
// Backend shape source: src/Sonarr.Api.V5/Manga/Blocklist/MangaBlocklistResource.cs
// (Phase 6 Plan 06-09). Field set is exact-mirror at the wire layer.
//
// Phase 8 cleanup: collapse with Blocklist when Tv/ deletes.
import ModelBase from 'App/ModelBase';

export interface MangaBlocklistSubresource {
  id: number;
  title?: string;
}

export interface MangaBlocklist extends ModelBase {
  mangaId: number;
  chapterIds?: number[];
  sourceTitle?: string;
  sourceKey?: string;
  releaseGuid?: string;
  date: string;
  reason?: string;
  source?: string;
  manga?: MangaBlocklistSubresource;
}

export default MangaBlocklist;
